R/pmcmc_state.R

```diff
@@ -337,11 +337,12 @@ pmcmc_state <- R6::R6Class(
         ## [state x particle x population x step]
         trajectories_state <-
           list_to_array(private$history_trajectories$get())
-        trajectories_state <- aperm(trajectories_state, c(1, 3, 2, 4))
+        trajectories_state <- aperm(trajectories_state, c(1, 4, 2, 3))
         rownames(trajectories_state) <- names(predict$index)
 
         data <- private$filter$inputs()$data
-        step <- c(data$step_start[[1]], data$step_end)
+        step_end <- data$step_end[seq_len(table(data$population)[1])]
```

Comment:
What is the role of table here?

Reply:
Gets the number of rows in each population

Reply:
is this not constant?

`tabulate` is likely faster and less magic